test/cmd/volumes.sh

```diff
@@ -6,33 +6,34 @@ set -o pipefail
 
 OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
 source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
 os::log::install_errexit
 
 # This test validates the 'volume' command
 
-oc create -f test/integration/fixtures/test-deployment-config.json
+os::cmd::expect_success 'oc create -f test/integration/fixtures/test-deployment-config.json'
 
-[ "$(oc volume dc/test-deployment-config --list | grep vol1)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol0 -m /opt5)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol2 --type=emptydir -m /opt)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol1 --type=secret --secret-name='$ecret' -m /data 2>&1 | grep overwrite)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol1 --type=emptyDir -m /data --overwrite)" ]
-[ "$(oc volume dc/test-deployment-config --add -m /opt 2>&1 | grep exists)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol2 -m /etc -c 'ruby' --overwrite 2>&1 | grep warning)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol2 -m /etc -c 'ruby*' --overwrite)" ]
-[ "$(oc volume dc/test-deployment-config --list --name=vol2 | grep /etc)" ]
-[ "$(oc volume dc/test-deployment-config --add --name=vol3 -o yaml | grep vol3)" ]
-[ "$(oc volume dc/test-deployment-config --list --name=vol3 2>&1 | grep 'not found')" ]
-[ "$(oc volume dc/test-deployment-config --remove 2>&1 | grep confirm)" ]
-[ "$(oc volume dc/test-deployment-config --remove --name=vol2)" ]
-[ ! "$(oc volume dc/test-deployment-config --list | grep vol2)" ]
-[ "$(oc volume dc/test-deployment-config --remove --confirm)" ]
-[ ! "$(oc volume dc/test-deployment-config --list | grep vol1)" ]
+os::cmd::expect_success_and_text 'oc volume dc/test-deployment-config --list' 'vol1'
+os::cmd::expect_success 'oc volume dc/test-deployment-config --add --name=vol0 -m /opt5'
+os::cmd::expect_success 'oc volume dc/test-deployment-config --add --name=vol2 --type=emptydir -m /opt'
+os::cmd::expect_failure_and_text "oc volume dc/test-deployment-config --add --name=vol1 --type=secret --secret-name='\$ecret' -m /data" 'overwrite'
```

Comment:
I was hoping @pravisankar would be able to say definitively that this was supposed to fail, as it may be the case that it was not supposed to, but was anyway.